### PR TITLE
Add async commit to Transaction, and allow reject rules to be provided

### PR DESCRIPTION
**Pull request by ghost**
Adds the ability to perform transactions asynchronously, and provide reject rules to the operations. Also moves to the tip of RAMCloud master so the nfapp build doesn't break.

### CODE REVIEW

**Review by OferMania — APPROVED** [review]
Looks good. Nothing here appears to be Stateless specific. Have you considered submitting these changes to RAMCloud repo?

**Comment by ghost** [comment]
> Looks good. Nothing here appears to be Stateless specific. Have you considered submitting these changes to RAMCloud repo?

Yeah we could probably submit this to ramcloud

**Comment by ghost** [comment]
Franza ran a test based on this branch, so there shouldn't be an issue committing this and bumping NFapp to use it.

File: patches/async-transaction.patch [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+     src/Status.h \
+     src/TestLog.h \
++    src/Transaction.h \
+     src/Transport.h \
+     src/Tub.h \
+     src/WireFormat.h \
+diff --git a/src/Transaction.cc b/src/Transaction.cc
+index c12867e3..8fb1f342 100644
+--- a/src/Transaction.cc
++++ b/src/Transaction.cc
```
Comment (vexingcodes) [inline patches/async-transaction.patch:39]: Several trailing whitespace issues on added lines in this fie.

**Review by vexingcodes — APPROVED** [review]
Fix the whitespace errors if you want. YOLO.

File: patches/async-transaction.patch (hunk 1 above)
Comment (ghost) [reply patches/async-transaction.patch:39]: donesies

**Pull request merged**